shared/devices/device-revoke/index.js

```diff
@@ -1,208 +1,140 @@
+const ActionButtons = ({onCancel, onSubmit}) => (
+  <Kb.Box2
+    direction={Styles.isMobile ? 'vertical' : 'horizontalReverse'}
+    fullWidth={Styles.isMobile}
     gap="tiny"
   >
-    <Button
-      fullWidth={isMobile}
+    <Kb.WaitingButton
+      fullWidth={Styles.isMobile}
       type="Danger"
-      onClick={props.waiting ? null : props.onSubmit}
-      label="Yes, revoke it"
-      disabled={props.waiting}
+      label="Yes, delete it"
+      waitingKey={Constants.waitingKey}
+      onClick={onSubmit}
     />
-    <Button fullWidth={isMobile} type="Secondary" onClick={props.onCancel} label="Cancel" />
-  </Box2>
+    <Kb.Button fullWidth={Styles.isMobile} type="Secondary" onClick={onCancel} label="Cancel" />
+  </Kb.Box2>
 )
 
-const DeviceRevoke = (props: Props) => (
-  <Box2 direction="vertical" fullHeight={true} fullWidth={true}>
-    <Box style={styles.deviceRevokeContainer}>
-      <Header name={props.name} type={props.type} />
-      <BodyText name={props.name} currentDevice={props.currentDevice} />
-      <EndangeredTLFList
-        endangeredTLFs={props.endangeredTLFs}
-        waiting={props.waiting}
-        style={styles.endangeredTLFList}
-      />
-      <ActionButtons onCancel={props.onCancel} onSubmit={props.onSubmit} waiting={props.waiting} />
-    </Box>
-  </Box2>
-)
+const DeviceRevoke = (props: Props) => {
+  const icon: Kb.IconType = {
+    backup: Styles.isMobile ? 'icon-paper-key-revoke-64' : 'icon-paper-key-revoke-48',
+    desktop: Styles.isMobile ? 'icon-computer-revoke-64' : 'icon-computer-revoke-48',
+    mobile: Styles.isMobile ? 'icon-phone-revoke-64' : 'icon-phone-revoke-48',
+  }[props.device.type]
 
-const styles = styleSheetCreate({
-  deviceRevokeContainer: {
-    ...globalStyles.flexBoxColumn,
-    alignItems: 'center',
-    flex: 1,
-    marginBottom: globalMargins.small,
-    marginLeft: globalMargins.small,
-    marginRight: globalMargins.small,
-  },
-  headerContainer: {
-    ...globalStyles.flexBoxColumn,
-    alignItems: 'center',
-    height: 112,
-    justifyContent: 'center',
-    marginBottom: globalMargins.small,
-  },
-  headerDeviceName: {
-    color: globalColors.red,
+  return (
+    <Kb.Box2
+      direction="vertical"
+      fullHeight={true}
+      fullWidth={true}
+      gap="small"
+      gapEnd={true}
+      style={styles.container}
+    >
+      <Kb.NameWithIcon icon={icon} title={props.device.name} titleStyle={styles.headerName} />
+      <Kb.Text type="Header" style={styles.centerText}>
+        Are you sure you want to revoke{' '}
+        {props.device.currentDevice ? (
+          'your current device'
+        ) : (
+          <Kb.Text type="Header" style={styles.italicName}>
+            {props.device.name}
+          </Kb.Text>
+        )}
+        ?
+      </Kb.Text>
+      <Kb.Box2 direction="vertical" style={styles.endangeredTLFContainer} fullWidth={Styles.isMobile}>
+        {props.waiting ? (
+          <Kb.ProgressIndicator />
```

Comment:
Could this be positioned absolutely somewhere? it makes things quickly jump around

Reply:
i'm removing it and adding text to the bottom
@cecileboucheron 
![screen shot 2018-07-31 at 10 53 07 am](https://user-images.githubusercontent.com/516435/43467575-f68c2c9a-94af-11e8-9af1-983fe98dce3f.png)